specs/altair/beacon-chain.md

```diff
@@ -276,16 +276,16 @@ def has_flag(flags: ParticipationFlags, flag_index: int) -> bool:
 def get_sync_committee_indices(state: BeaconState, epoch: Epoch) -> Sequence[ValidatorIndex]:
     """
     Return the sequence of sync committee indices (which may include duplicate indices)
-    for a given ``state`` and ``epoch``.
-
-    Note: This function is not stable during a sync committee period as
-    a validator's effective balance may change enough to affect the sampling.
+    for a given ``state`` and ``epoch`` at a sync committee period boundary.
     """
     MAX_RANDOM_BYTE = 2**8 - 1
-    base_epoch = Epoch((max(epoch // EPOCHS_PER_SYNC_COMMITTEE_PERIOD, 1) - 1) * EPOCHS_PER_SYNC_COMMITTEE_PERIOD)
-    active_validator_indices = get_active_validator_indices(state, base_epoch)
+
+    if epoch != ALTAIR_FORK_EPOCH or epoch != ALTAIR_FORK_EPOCH + EPOCHS_PER_SYNC_COMMITTEE_PERIOD:
+        assert epoch % EPOCHS_PER_SYNC_COMMITTEE_PERIOD == 0
+
+    active_validator_indices = get_active_validator_indices(state, epoch)
     active_validator_count = uint64(len(active_validator_indices))
-    seed = get_seed(state, base_epoch, DOMAIN_SYNC_COMMITTEE)
+    seed = get_seed(state, epoch, DOMAIN_SYNC_COMMITTEE)
```

Comment:
this ensures the `epoch` is aligned or part of the special case at the altair fork epoch.

it also removes the prior "base epoch" computation with the only substantial change being that the committees computed right at the fork will have "non-aligned" `seed` selection from the state. imo fine given the simplification it enables